File: apple-and-orange/apple-and-orange.go
package appleandorange

import "fmt"

func countApplesAndOranges(s int32, t int32, a int32, b int32, apples []int32, oranges []int32) {
	fmt.Println(hitCount(s, t, a, apples))
	fmt.Println(hitCount(s, t, b, oranges))
}

func hitCount(
	targetStart int32,
	targetEnd int32,
	hitterStart int32,
	hitterDistances []int32,
) int32 {
	var result int32 = 0

	for _, d := range hitterDistances {
		hitterEnd := hitterStart + d
		isHit := hitterEnd >= targetStart && hitterEnd <= targetEnd
		if isHit {
			result++
		}
	}

	return result
}

File: apple-and-orange/AppleAndOrange.java
import java.util.List;

public class AppleAndOrange {

  public static void countApplesAndOranges(int s, int t, int a, int b, List<Integer> apples, List<Integer> oranges) {
    System.out.println(getHitCount(s, t, a, apples));
    System.out.println(getHitCount(s, t, b, oranges));
  }

  private static long getHitCount(
    int targetStart,
    int targetEnd,
    int hitterStart,
    List<Integer> hitterDistances
  ) {
    return hitterDistances
      .stream()
      .map(hitterDistance -> hitterDistance + hitterStart)
      .filter(hitterEnd -> 
        hitterEnd >= targetStart
        &&
        hitterEnd <= targetEnd
      )
      .count();
  }

}
